test/index.js

```diff
@@ -0,0 +1,450 @@
+        message: 'fatal unknown',
+        severity: DiagnosticSeverity.Information
+      }
+    ]
+  })
+
+  t.end()
+})
+
+test('onDidChangeContent has ruleId', async (t) => {
+  const uri = String(pathToFileURL('test.md'))
+  const diagnostics = await getDiagnostic(uri, 'has ruleId')
+
+  t.deepEquals(diagnostics, {
+    uri,
+    version: 0,
+    diagnostics: [
+      {
+        code: 'test-rule',
+        range: {start: {line: 0, character: 0}, end: {line: 0, character: 0}},
+        message: 'has ruleId',
+        severity: DiagnosticSeverity.Warning
+      }
+    ]
+  })
+
+  t.end()
+})
+
+test('onDidChangeContent has source', async (t) => {
+  const uri = String(pathToFileURL('test.md'))
+  const diagnostics = await getDiagnostic(uri, 'has source')
+
+  t.deepEquals(diagnostics, {
+    uri,
+    version: 0,
+    diagnostics: [
+      {
+        range: {start: {line: 0, character: 0}, end: {line: 0, character: 0}},
+        message: 'has source',
+        source: 'test-source',
+        severity: DiagnosticSeverity.Warning
+      }
+    ]
+  })
+
+  t.end()
+})
+
+test('onDidChangeContent has url', async (t) => {
+  const uri = String(pathToFileURL('test.md'))
+  const diagnostics = await getDiagnostic(uri, 'has url')
+
+  t.deepEquals(diagnostics, {
+    uri,
+    version: 0,
+    diagnostics: [
+      {
+        codeDescription: {
+          href: 'https://example.com'
+        },
+        range: {start: {line: 0, character: 0}, end: {line: 0, character: 0}},
+        message: 'has url',
+        severity: DiagnosticSeverity.Warning
+      }
+    ]
+  })
+
+  t.end()
+})
+
+test('onDidChangeContent has error', async (t) => {
+  const uri = String(pathToFileURL('test.md'))
+  const diagnostics = await getDiagnostic(uri, 'has error')
+
+  t.deepEquals(diagnostics, {
+    uri,
+    version: 0,
+    diagnostics: [
+      {
+        range: {start: {line: 0, character: 0}, end: {line: 0, character: 0}},
+        message: 'Test error',
+        severity: DiagnosticSeverity.Error
+      }
+    ]
+  })
+
+  t.end()
+})
+
+// test('onDidChangeConfiguration', async (t) => {
+//   const connection = createMockConnection()
+//   const documents = new TextDocuments(TextDocument)
+//   const diagnosticsPromise = new Promise((resolve) => {
+//     const sendDiagnostics = /** @type import('sinon').SinonStub */ (
+//       connection.sendDiagnostics
+//     )
+//     sendDiagnostics.callsFake(() => {
+//       if (sendDiagnostics.callCount === 2) {
+//         resolve([
+//           sendDiagnostics.firstCall.firstArg,
+//           sendDiagnostics.lastCall.firstArg
+//         ])
+//       }
+//     })
+//   })
+//   const uri1 = String(pathToFileURL('test1.md'))
+//   const uri2 = String(pathToFileURL('test2.md'))
+
+//   Object.defineProperty(documents, 'all', {
+//     value: () => [
+//       TextDocument.create(uri1, 'text', 0, 'has ruleId'),
+//       TextDocument.create(uri2, 'text', 0, 'has source')
```

Comment:
This test should be uncommented. Anyway, this test fails because `unified-engine` only appears to run the plugins on the first file passed. Could someone find out what’s wrong?

In practice this means only one file gets checked when the editor is started with multiple markdown files open.

Reply:
Apparently this isn’t necessary at all.